pkg/router/router_test.go

```diff
@@ -0,0 +1,131 @@
+package router
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/yaml"
+)
+
+func TestPorts(t *testing.T) {
+	tests := []struct {
+		name              string
+		specifiedPorts    []corev1.ServicePort
+		expectedPortsFile string
+	}{
+		{
+			name:              "default ports",
+			expectedPortsFile: "default-ports.yaml",
+		},
+		{
+			name: "additional ports",
+			specifiedPorts: []corev1.ServicePort{
+				{
+					Name: "additional port",
+					Port: 4308,
+				},
+				{
+					Name: "additional port with target port",
+					Port: 1337,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 20,
+					},
+				},
+			},
+			expectedPortsFile: "add-ports.yaml",
+		},
+		{
+			name: "modified ports with additional ports",
+			specifiedPorts: []corev1.ServicePort{
+				{
+					Name: "http",
+					Port: 5555,
+				},
+				{
+					Name: "rw-default",
+					Port: 6666,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 30,
+					},
+				},
+				{
+					Name: "additional port",
+					Port: 4308,
+				},
+				{
+					Name: "additional port with target port",
+					Port: 1337,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 20,
+					},
+				},
+			},
+			expectedPortsFile: "mod-add-ports.yaml",
+		},
+		{
+			name: "modified port with default targetPort",
+			specifiedPorts: []corev1.ServicePort{
+				{
+					Name: "rw-default",
+					Port: 6666,
+				},
+			},
+			expectedPortsFile: "mod-def-targetport-ports.yaml",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := ports(tt.specifiedPorts)
+			data, err := yaml.Marshal(got)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = os.WriteFile(filepath.Join("testdata", "ports", tt.expectedPortsFile), data, 0666)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Why are we doing that? I think the test is passing for the wrong reason when this code is executed. Essentially we are overwritting the expectation file with the data we got from the `ports` function.